lib/braintrust/trace/span_processor.rb

```diff
@@ -73,6 +76,29 @@ def get_parent_from_context(parent_context)
         # Return the parent attribute from the parent span
         parent_span.attributes&.[](PARENT_ATTR_KEY)
       end
+
+      # Determine if a span should be forwarded to the wrapped processor
+      # based on configured filters
+      def should_forward_span?(span)
+        # Always keep root spans (spans with no parent)
+        # Check if parent_span_id is the invalid/zero span ID
+        is_root = span.parent_span_id == OpenTelemetry::Trace::INVALID_SPAN_ID
+        return true if is_root
+
+        # If no filters, keep everything
+        return true if @filters.empty?
+
+        # Apply filters in order - first non-zero result wins
+        @filters.each do |filter|
+          result = filter.call(span)
+          return true if result > 0  # Keep span
+          return false if result < 0 # Drop span
```

Comment:
Do we make any guarantees about what we do if filters contradict each other? The implementation has a "first one wins" approach, which seems sane. Just wondering if we guarantee that behavior or if it's undefined?

Reply:
first one wins!